src/components/home/modals/AliasAddress.vue

```diff
@@ -0,0 +1,303 @@
+                      v-model="curAddress"
+                      :readonly="from3rdParty"
+                      aria-describedby="inputLiveFeedback"
+                      :state="isValidAddress"
+                      onfocus="this.select()">
+        </b-form-input>
+        <b-form-invalid-feedback id="inputLiveFeedback"
+                                 v-if="!isExistedAddress">
+          Please input your valid address!
+        </b-form-invalid-feedback>
+      </b-form-group>
+      <b-form-group label="Alias Name"
+                    label-for="alias-input">
+        <b-form-input id="alias-input"
+                      class="amount-input"
+                      v-model="curAlias"
+                      :readonly="from3rdParty"
+                      aria-describedby="inputLiveFeedback"
+                      :state="isValidAlias"
+                      onfocus="this.select()">
+        </b-form-input>
+        <b-form-invalid-feedback id="inputLiveFeedback"
+                                 v-if="!isAliasLessEight">
+          Please enter an alias no more than 8 characters.
+        </b-form-invalid-feedback>
+        <b-form-invalid-feedback id="inputLiveFeedback"
+                                 v-if="isExistedAlias">
+          You have set this alias to another address, please change it to another！
+        </b-form-invalid-feedback>
+      </b-form-group>
+      <b-row>
+        <b-col class="col-lef">
+          <b-button
+            class="btn-cancel"
+            block
+            variant="light"
+            size="lg"
+            @click="closeModal">Close
+          </b-button>
+        </b-col>
+        <b-col class="col-rit">
+          <b-button
+            block
+            class="btn-add"
+            variant="warning"
+            size="lg"
+            @click="addAlias">Add
+          </b-button>
+        </b-col>
+      </b-row>
+    </div>
+  </b-modal>
+</template>
+
+<script>
+import Vue from 'vue'
+export default {
+    name: 'AliasAddress',
+    props: {
+        addresses: {
+            type: Object,
+            require: true,
+            default: function() {}
+        },
+        address: {
+            type: String,
+            require: true,
+            default: ''
+        }
+    },
+    data() {
+        return {
+            curAlias: '',
+            curAddress: '',
+            aliasNum: 3,
+            aliasAddresses: {}
+        }
+    },
+    created() {
+        this.aliasAddresses = JSON.parse(window.localStorage.getItem(this.address)).alias
+    },
+    computed: {
+        isValidAddress() {
+            if (!this.curAddress) {
+                return void 0
+            }
+            return this.isExistedAddress
+        },
+        isExistedAddress() {
+            return this.curAddress in this.addresses
+        },
+        isExistedAlias() {
+            let alias = JSON.parse(window.localStorage.getItem(this.address)).alias
+            for (let key in alias) {
```

Comment:
alias could be null here. check alias is null or not before do `for` loop.
<img width="581" alt="Screenshot 2020-01-21 at 1 37 41 PM" src="https://user-images.githubusercontent.com/41148850/72778436-f777d800-3c53-11ea-97b0-9b8173bfe3ad.png">